x-pack/plugins/maps/public/classes/sources/es_search_source/top_hits/update_source_editor.tsx

```diff
@@ -0,0 +1,233 @@
+              <FormattedMessage
+                id="xpack.maps.esSearch.tooltipsTitle"
+                defaultMessage="Tooltip fields"
+              />
+            </h5>
+          </EuiTitle>
+
+          <EuiSpacer size="m" />
+
+          <TooltipSelector
+            tooltipFields={this.props.tooltipFields}
+            onChange={this._onTooltipPropertiesChange}
+            fields={this.state.sourceFields}
+          />
+        </EuiPanel>
+        <EuiSpacer size="s" />
+
+        <EuiPanel>
+          <EuiTitle size="xs">
+            <h5>
+              <FormattedMessage
+                id="xpack.maps.source.topHitsPanelLabel"
+                defaultMessage="Top hits"
+              />
+            </h5>
+          </EuiTitle>
+
+          <EuiSpacer size="m" />
+
+          <TopHitsForm
+            indexPatternId={this.props.indexPatternId}
+            isColumnCompressed={true}
+            onChange={this.props.onChange}
+            termFields={this.state.termFields}
+            topHitsSplitField={this.props.topHitsSplitField}
+            topHitsSize={this.props.topHitsSize}
+          />
+
+          <EuiFormRow
+            label={i18n.translate('xpack.maps.source.esTopHitsSearch.sortFieldLabel', {
+              defaultMessage: 'Sort field',
+            })}
+            display="columnCompressed"
+          >
+            <SingleFieldSelect
+              placeholder={i18n.translate('xpack.maps.source.esSearch.sortFieldSelectPlaceholder', {
+                defaultMessage: 'Select sort field',
+              })}
+              value={this.props.sortField}
+              onChange={this._onSortFieldChange}
+              fields={this.state.sortFields}
+              compressed
+            />
+          </EuiFormRow>
+
+          <EuiFormRow
+            label={i18n.translate('xpack.maps.source.esTopHitsSearch.sortOrderLabel', {
+              defaultMessage: 'Sort order',
+            })}
+            display="columnCompressed"
+          >
+            <EuiSelect
+              disabled={!this.props.sortField}
+              options={[
+                {
+                  text: i18n.translate('xpack.maps.source.esSearch.ascendingLabel', {
+                    defaultMessage: 'ascending',
+                  }),
+                  value: SortDirection.asc,
+                },
+                {
+                  text: i18n.translate('xpack.maps.source.esSearch.descendingLabel', {
+                    defaultMessage: 'descending',
+                  }),
+                  value: SortDirection.desc,
+                },
+              ]}
+              value={this.props.sortOrder}
+              onChange={this._onSortOrderChange}
+              compressed
+            />
+          </EuiFormRow>
+
+          <EuiFormRow>
+            <EuiSwitch
+              label={i18n.translate('xpack.maps.source.esSearch.extentFilterLabel', {
+                defaultMessage: 'Dynamically filter for data in the visible map area',
```

Comment:
What about renaming this, to indicate more the actual impact.

e.g.

`only take into account documents within the visible map-extent to determine the top hit`

^ this is a consequence of the implementation right now. Tracks-layer does this better. (doesn't need to be fixed here imho)

Reply:
ya, lets leave this for another PR.